internal/docker/builder.go

```diff
@@ -612,9 +614,18 @@ func deployImage(
 	return d, nil
 }
 
-// CreateNetwork creates a docker network and returns its id.
+// CreateNetworkIfNotExists creates a docker network and returns its id.
 // ID is guaranteed not to be empty when err == nil
-func CreateNetwork(docker *client.Client, blueprintName string) (networkID string, err error) {
+func CreateNetworkIfNotExists(docker *client.Client, blueprintName string) (networkID string, err error) {
+	nws, err := docker.NetworkList(context.Background(), types.NetworkListOptions{
+		Filters: label("complement_blueprint=" + blueprintName),
+	})
+	if err != nil {
+		return "", fmt.Errorf("%s: failed to list networks. %w", blueprintName, err)
+	}
+	if len(nws) > 0 {
+		return nws[0].ID, nil
+	}
```

Comment:
I think just a comment here explaining what this section is doing could go a long way.

I'm also not the biggest fan of abbreviated variable names, though I see that's consistent with the existing code.

Reply:
I shall point you to https://github.com/golang/go/wiki/CodeReviewComments#variable-names for why we do this.